content/posts/0006-build-plan-buildpack.md

```diff
@@ -0,0 +1,93 @@
+have all of the dependencies that it provides, required by buildpacks that come
+later in the build order. If either of these conditions is not satisfied
+detection will fail for that buildpack. A more in-depth breakdown of the Build
+Plan can be found in the [CNB
+Spec](https://github.com/buildpacks/spec/blob/main/buildpack.md).
+
+### How does the Build Plan buildpack work?
+
+The Build Plan buildpack works by adding a `plan.toml`, which mirrors the
+[requires portion of Build
+Plan](https://github.com/buildpacks/spec/blob/main/buildpack.md#build-plan-toml),
+to the root of your app. The buildpack will read the `plan.toml` and make all
+of the requirements listed in it for you. This means that if you add a
+`plan.toml` to the root of your app and then adds the Build Plan buildpack to
+the end of your  buildpack group, you will be able to make any arbitrary
+requirement that you want. Furthermore, if you are using the Build Plan
+buildpack with Paketo Buildpacks it will allow you to specify when you want the
+dependency to be present whether that be during build, at launch, or both. You
+can specify this for Paketo Buildpacks because they respect two flags that go
+in the `metadata` field of a requirement. Those two flags are `build` and
+`launch`, `build` makes the dependency available during the build phase of
+subsequent buildpacks in the order, and `launch` makes the dependency available
+in the final running image.
+
+### Why would I want this?
+
+The place where the Build Plan buildpack is used heavily right now is during
+the integration testing of dependency buildpacks. Let's use a [Go Dist
+buildpack integration
+app](https://github.com/paketo-buildpacks/go-dist/tree/main/integration/testdata/default_app)
+as an example. The `plan.toml` for that app looks as follow:
+```
+[[requires]]
+  name = "go"
+
+  [requires.metadata]
+    launch = true
+```
+The Build Plan buildpack is used here so that we don't need to use either the
+Go Mod Vendor or the Go Build buildpacks to require the `go` dependency. We can
+also force the `go` dependency, which in this case is the Go compiler, to be
+present in the final running image, which it normally is absent from. This
+allows us to run various `go` commands to ensure that it is installed properly
+without relying on any other buildpacks in the Go language family.
+
+There are other applications where the use of the Build Plan buildpack would
+enable a build that would be hard if not impossible to detect automatically, or
+to make an image as small as possible. A great example of this would be
+something like an app that has a simple Nodejs frontend and has a Go backend.
+You could add the Node Engine and Build Plan buildpack to the order grouping of
+your build and write a `plan.toml` that required the node engine during launch.
+This would force the Node Engine buildpack's detection to pass and the build
+phase would install the node engine into the image and make it available during
+build.
```

Comment:
I feel like this example should be the star here, since it's the most likely scenario (I'd imagine) a reader of this blog post would be interested in

Reply:
I think the only issue with making it the star example is that is it a little complicated. I would rather mention it as an option than feature it.

Reply:
Bump